gapic-common/lib/gapic/grpc/service_stub/channel_pool.rb

```diff
@@ -0,0 +1,113 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "grpc"
+require "googleauth"
+require "gapic/config"
+
+
+module Gapic
+  class ServiceStub
+    ##
+    # Gapic gRPC ServiceStub ChannelPool
+    #
+    # This class wraps multiple channels for sending RPCs.
+    #
+    class ChannelPool
+      ##
+      # Initialize an instance of ServiceStub::ChannelPool
+      #
+      def initialize grpc_stub_class, endpoint:, credentials:, channel_args: nil, interceptors: nil, config: nil
+        if credentials.is_a? ::GRPC::Core::Channel
+          raise ArgumentError, "Can't create a channel pool with GRPC::Core::Channel as credentials"
+        end
+
+        @grpc_stub_class = grpc_stub_class
+        @endpoint = endpoint
+        @credentials = credentials
+        @channel_args = channel_args
+        @interceptors = interceptors
+        @config = config || Configuration.new
+
+        yield @config if block_given?
```

Comment:
I don't think we need this. Currently the only thing that constructs a ChannelPool is a ServiceStub, and it doesn't pass a block. We don't expect users to construct ChannelPool directly.

Reply:
Removed the yield.